cli: remove newline insertion after template expansion

higgsd opened the pull request:
In graph mode, a newline is appended whether or not there is content (diffs, file listings, op changes) following the expanded template.  This does not happen in no-graph mode, causing unwanted concatenation, which is particularly problematic for diff headers.

[several edits later]

After several attempts at addressing this, the approved solution is to remove the newline addition.  Built-in templates are unaffected, but users get exactly what they ask for when using a custom template.

# Checklist

If applicable:

- [x] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (`README.md`, `docs/`, `demos/`)
- [ ] I have updated the config schema (`cli/src/config-schema.json`)
- [x] I have added/updated tests to cover my changes

yuja commented on cli/src/commands/evolog.rs:
```diff
@@ -210,21 +209,29 @@ pub(crate) fn cmd_evolog(
                     )
                     .block_on()?;
             }
+            write!(raw_output, "{}", &String::from_utf8_lossy(&buffer))?;
```
just `.write_all(&buffer)`

higgsd commented:
> > perhaps the default templates should be rewritten without newlines, as templates seem more composable without newlines
> 
> Since `--no-graph` output shouldn't insert newlines implicitly, the builtin templates will have to be terminated by newline. 

I see the primary use-case of `--no-graph` as rendering disjoint nodes without a graph, and the ability to produce horizontal output from multiple commits has rather limited utility.  The goal of my PR was just to improve the existing rendering - changing the builtin templates or forcing people to update their custom templates is a tall order.  As such, anything beyond this would be a different FR/PR and I am not sufficiently opinionated to iterate further at this time.

> BTW, someone asked inserting blank lines before/after the `--patch` output. I don't know whether this would be better, but it will mitigate bad formatting of patch contents.

The builtin diff rendering is good at conserving vertical whitespace, and I feel like extra newlines wouldn't be very popular. It would address the problem though.

All just my 2c, though.

higgsd commented:
I found a few more cases where templates without newlines were not separated from their "content", and added some test cases for them.

Then I removed a bunch of inconsistent newline usage, and re-added unconditional newlines in as few spots as necessary.  Let me know what you think.

glehmann commented:
The behavior has changed with `--summary`. I think this is linked to this PR: unlike before, the summary begins just after template content:

<img width="578" height="260" alt="image" src="https://github.com/user-attachments/assets/01f477f7-3cd2-468f-9975-52216946e2ba" />

Is it the new expected behavior?
